objects/src/constants.rs

```diff
@@ -0,0 +1,63 @@
+/// Depth of the account database tree.
+pub const ACCOUNT_TREE_DEPTH: u8 = 64;
+
+/// The depth of the Merkle tree used to commit to notes produced in a block.
+pub const NOTE_TREE_DEPTH: u8 = 20;
+
+/// The maximum number of assets that can be stored in a single note.
+pub const MAX_ASSETS_PER_NOTE: usize = 256;
+
+/// The maximum number of inputs that can accompany a single note.
+///
+/// The value is set to 128 so that it can be represented using as a single byte while being
+/// evenly divisible by 8.
+pub const MAX_INPUTS_PER_NOTE: usize = 128;
+
+/// The maximum number of notes that can be consumed by a single transaction.
+pub const MAX_INPUT_NOTES_PER_TX: usize = 1023;
+
+/// The maximum number of new notes created by a single transaction.
+pub const MAX_OUTPUT_NOTES_PER_TX: usize = 4096;
+
+/// The minimum proof security level used by the Miden prover & verifier.
+pub const MIN_PROOF_SECURITY_LEVEL: u32 = 96;
+
+// TRANSACTION BATCH
+// ================================================================================================
+
+/// The depth of the Sparse Merkle Tree used to store created notes in a single batch.
+///
+/// A single note uses two leaves in the SMT. The even leaf is used to store the note's id, the odd
+/// leaf is used to store the note's metadata.
+pub const BATCH_CREATED_NOTES_SMT_DEPTH: u8 = 13;
+
+/// The maximum number of notes that can be created in a single batch.
+///
+/// Because the SMT used in a batch has fixed depth, and each note takes two leaves, the maximum
+/// number of notes is the number of leaves in the SMT.
+pub const BATCH_MAX_NOTES: usize = 2_usize.pow((BATCH_CREATED_NOTES_SMT_DEPTH - 1) as u32);
+
+/// The maximum number of transaction in a single batch.
+pub const BATCH_MAX_TRANSACTIONS: usize = BATCH_MAX_NOTES / MAX_OUTPUT_NOTES_PER_TX;
+
+// BLOCK
+// ================================================================================================
+
+/// The depth of the Sparse Merkle Tree used to store a batch's note SMT.
+///
+/// This value can be interpreted as:
+///
+/// - The depth of a tree with the leaves set to a batch created note SMT root.
+/// - The level at which the batches create note SMTs are merged, creating a new SMT with this many
+///   additional new levels.
+pub const BLOCK_CREATED_NOTES_SMT_BATCH_DEPTH: u8 = 8;
+
+/// The final depth of the Sparse Merkle Tree used to store all notes created in a block.
+pub const BLOCK_CREATED_NOTES_SMT_DEPTH: u8 =
+    BATCH_CREATED_NOTES_SMT_DEPTH + BLOCK_CREATED_NOTES_SMT_BATCH_DEPTH;
+
+/// Maximum number of batches that can be inserted into a single block.
+pub const BLOCK_MAX_BATCHES: usize = 2_usize.pow(BLOCK_CREATED_NOTES_SMT_BATCH_DEPTH as u32);
+
+/// The block height of the genesis block
+pub const GENESIS_BLOCK: u32 = 0;
```

Comment:
I personally really like to follow a standard for constants, like the one above using `BLOCK_` and `BATCH_`. It makes it easier to manipulate code and see related constants in imports (because they are lexicographically close to each other).

I didn't do this here though, because this name is used in the `miden-node` repo. The same for the transaction constants that already lived here.